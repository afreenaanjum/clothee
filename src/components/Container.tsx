import React from 'react';
import {StyleSheet, View} from 'react-native';
import {IContainerProps} from '../interfaces/interface';
const Container: React.FC<IContainerProps> = ({
  children,
  style,
}: IContainerProps) => {
  return <View style={[styles.container, style]}>{children}</View>;
};

const styles = StyleSheet.create({container: {flex: 1}});

export default Container;
