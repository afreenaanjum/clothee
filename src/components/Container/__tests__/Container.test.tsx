import 'react-native';
import {Text} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../Container';

describe('Center widget', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Container children={<Text>Container</Text>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when no children', () => {
    const tree = renderer.create(<Container children={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
