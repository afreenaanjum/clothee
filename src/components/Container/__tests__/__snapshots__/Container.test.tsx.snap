// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center widget renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text>
    Container
  </Text>
</View>
`;

exports[`Center widget renders correctly when no children 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
